astroquery/mast/discovery_portal.py

```diff
@@ -406,8 +407,10 @@ def build_filter_set(self, column_config_name, service_name=None, **filters):
             # Get the column type and separator
             col_info = caom_col_config.get(colname)
             if not col_info:
-                warnings.warn("Filter {} does not exist. This filter will be skipped.".format(colname), InputWarning)
-                continue
+                closest_match = difflib.get_close_matches(colname, caom_col_config.keys(), n=1)[0]
+                error_msg = f"Filter '{colname}' does not exist. Did you mean '{closest_match}'?" if closest_match \
+                    else f"Filter '{colname}' does not exist."
+                raise InvalidQueryError(error_msg)
```

Comment:
Given this changes the warning to an error, please add a short changelog for the users.

Reply:
Ok, tests should pass now, sorry about that!

Reply:
No worries, this is what review is for.